OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -124,7 +125,11 @@ public Aircraft(ActorInitializer init, AircraftInfo info)
 			IsPlane = !info.CanHover;
 		}
 
-		public void Created(Actor self) { um = self.TraitOrDefault<UpgradeManager>(); }
+		public void Created(Actor self)
+		{
+			um = self.TraitOrDefault<UpgradeManager>();
+			speedModifiers = self.TraitsImplementing<ISpeedModifier>().ToArray().Select(sm => sm.GetSpeedModifier());
```

Comment:
Yes, the `IEnumerable<int>` created by `.Select(sm => sm.GetSpeedModifier())` holds a reference to the unnamed `ISpeedModifier[]` created by `.ToArray()`.